consensus/config/src/committee.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use multiaddr::Multiaddr;
+use serde::{Deserialize, Serialize};
+
+use crate::{NetworkPublicKey, ProtocolPublicKey};
+
+/// Committee of the consensus protocol is updated each epoch.
+pub type Epoch = u64;
+
+/// Each authority is uniquely identified by its AuthorityIndex in the Committee.
+/// AuthorityIndex is between 0 (inclusive) and the total number of authorities (exclusive).
+pub type AuthorityIndex = u32;
+
+/// Voting power of an authority, roughly proportional to the actual amount of Sui staked
+/// by the authority.
+/// Total stake / voting power of all authorities should sum to 10,000.
+pub type Stake = u64;
+
+/// Network information of one authority in the committee.
+#[derive(Serialize, Deserialize)]
+pub struct NetworkInfo {
```

Comment:
To be honest I would prefer if we could bundle all the authority information (network, stake, id etc) under an `Authority` struct as we [do in Narwhal](https://github.com/MystenLabs/sui/blob/cae95c6bb9ef163e99a8f6f95d7e5bacc85aded6/narwhal/config/src/committee.rs#L20)  . I see that during initialisation it might not be that delicate, but in the end we can avoid fragmenting our Committee with methods for asking every authority detail .

Reply:
sg.